tests/test_extensions/test_blocks/test_captions.py

```diff
@@ -226,6 +226,91 @@ def test_manual_prepend(self):
             True
         )
 
+    def test_caption_inline_id(self):
+        """Test caption with inline shorthand ID."""
+
+        self.check_markdown(
+            R'''
+            Paragraph
+            /// figure-caption | #custom-id
+            Caption text.
+            ///
+            ''',
+            R'''
+            <figure id="custom-id">
+            <p>Paragraph</p>
+            <figcaption>
+            <p>Caption text.</p>
+            </figcaption>
+            </figure>
+            ''',
+            True
+        )
+
+    def test_caption_inline_id_prepend(self):
+        """Test caption with inline shorthand ID and prepend marker."""
+
+        self.check_markdown(
+            R'''
+            Text
+            /// figure-caption | < #custom-prepend
+            Prepended caption.
+            ///
+            ''',
+            R'''
+            <figure id="custom-prepend">
+            <figcaption>
+            <p>Prepended caption.</p>
+            </figcaption>
+            <p>Text</p>
+            </figure>
+            ''',
+            True
+        )
+
+    def test_table_caption_inline_id(self):
+        """Test table caption with inline shorthand ID."""
+
+        self.check_markdown(
+            R'''
+            Paragraph
+            /// table-caption | #custom-table
+            Table caption text.
+            ///
+            ''',
+            R'''
+            <figure id="custom-table">
+            <p>Paragraph</p>
+            <figcaption>
+            <p>Table caption text.</p>
+            </figcaption>
+            </figure>
+            ''',
+            True
+        )
+
+    def test_caption_inline_id_attrs_override(self):
+        """Test inline shorthand ID does not override attrs ID."""
```

Comment:
Surround 'attrs' in backticks so it will be ignored by the spellchecker: `` `attrs` ``.